doc/proposals/ansible-operator.md

```diff
@@ -0,0 +1,78 @@
+
+### Goals
+
+The goal of the Ansible Operator will be to create a fully functional framework for Ansible developers to create operators. It will also expose a library for golang users to use ansible in their operator if they so choose. These two goals in conjunction will allow users to select the best technology for their project or skillset. 
+
+### New Operator Type
+
+This proposal creates a new type of operator called `ansible`.  The new type is used to tell the tooling to act on that type of operator. 
+
+### Package Structure
+Packages will be added to the operator-sdk. These packages are designed to be usable by the end user if they choose to and should have a well documented public API. The proposed packages are:
+* /operator-sdk/pkg/ansible/controller
+  * Will contain the ansible operator controller.
+  * Will contain a exposed reconciler. But the default `Add` method will use this reconciler.
+* /operator-sdk/pkg/ansible/runner
+  * Contains the runner types and interfaces
+  * Implementation is behind an internal package (/operator-sdk/pkg/ansible/runner/internal)
+  * New Methods are exposed and are the main way a user interacts with the package
+  * Runner interface for running ansible from the operator.
+  * NewForWatchers - the method that returns a map of GVK to Runner types based on the watchers file.
+  * NewPlaybookRunner - the method that returns a new Runner for a playbook.
+  * NewRoleRunner - the method that returns a new Runner for a role.
+  * This contains the events API code and public methods. Implementation should probably be in the internal package. The events API is used for recieving events from ansible runner.
+
+* /operator-sdk/pkg/ansible/proxy
+  * This is a reverse proxy for the kubernetes API that is used for owner reference injection.
+* /operator-sdk/pkg/ansible/proxy/kubeconfig
+  * Code needed to generate the kubeconfig for the proxy.
+* /operator-sdk/pkg/ansible/events
+  * Package for event handlers from ansible runner events.
+  * Default has only the event logger.
+
+
+### Commands
+We are adding and updating existing commands to accommodate the ansible operator.  Changes to the `cmd` package as well as changes to the generator are needed.
+
+`operator-sdk new <project-name> --type ansible --kind <kind> --api-version <group/version>`  This will be a new generation command under the hood. We will:
+* Create a new ansible role in the roles directory
+* Create a new watchers file. The role and GVK are defaulted based on input to the command.
+* A CRD is generated. This CRD does not have any validations defined.
+* A dockerfile is created using the watchers file and the ansible role with the base image being the ansible operator base image.
+
+The resulting structure should be
+```
+|- Dockerfile
+|
+|- roles
+\ | - <kind>
+|  \ | - generated ansible role
+|
+| - watchers.yaml
```

Comment:
@shawn-hurley shouldn't this be `watches.yaml`?

Reply:
yes, oops 🤦‍♂️

Reply:
cool.. thought it was me :)